Exercise/src/my_string.cpp

```diff
@@ -173,29 +173,229 @@ Exercise_1::CMyString& Exercise_1::CMyString::operator=(const Exercise_1::CMyStr
 }
 
 // Destructor
-Exercise_1::CMyString::~CMyString()
+MyStructs::CMyString::~CMyString()
 {
 	if(m_szData != nullptr)
 	{
 		delete[] m_szData;
 	}
 }
 
-Exercise_1::CMyString& Exercise_1::CMyString::operator=(const char* cpszCharsSequence)
+void MyStructs::CMyString::Delete(size_t nPosition, size_t nCharactersCount) noexcept
+{
+	if(IsOutOfBounds(nPosition, nCharactersCount))
+	{
+		return;
+	}
+	if(nCharactersCount == size() - 1) 
+	{
+		delete[] m_szData;
+		m_szData = nullptr;
+		m_nSize = 0;
+		return;
+	}
+
+	const size_t cnOldSize = m_nSize;
+	// Calculating new size with getting in mind that user can delete zero terminant
+	m_nSize -= nCharactersCount - IsReachedTerminator(nPosition, nCharactersCount);
+
+
+	char *szOldString = m_szData; 
+	TryToAllocate(m_nSize);
+	if(m_szData == nullptr)
+	{
+		return;
+	}
+	
+	size_t nOldStringPosition = 0;
+	size_t nNewStringPosition = 0;
+
+	// Moving to cnOldSize - 2, to manualy add '\0' in the end in case we deleted it
+	while(nOldStringPosition < cnOldSize - 1) 
+	{
+		if(nOldStringPosition == nPosition)
+		{
+			nOldStringPosition += nCharactersCount;
+			continue;
+		}
+		
+		m_szData[nNewStringPosition] = szOldString[nOldStringPosition];
+		++nOldStringPosition;
+		++nNewStringPosition;
+	}
+
+	m_szData[m_nSize - 1] = '\0';
+
+	delete[] szOldString;
+}
+
+MyStructs::CMyString MyStructs::CMyString::Substring(size_t nPosition, size_t nCharactersCount) const noexcept
+{
+	CMyString substring{nullptr}; 
+	if(nPosition >= m_nSize || nCharactersCount == 0 || 
+		m_nSize - nPosition < nCharactersCount)
+	{
+		return substring;
+	}
+	substring.TryToAllocate(nCharactersCount);
+	if(substring.m_szData == nullptr)
+	{
+		return substring;
+	}
+
+	const size_t cnEndPosition = nPosition + nCharactersCount;
+	for(size_t nCurrentIndex = nPosition; nCurrentIndex < cnEndPosition; nCurrentIndex++)
+	{
+		substring.m_szData[nCurrentIndex - nPosition] = m_szData[nCurrentIndex];
+	}
+
+	return substring;
+}
+std::optional<size_t> MyStructs::CMyString::Find(const char *cszStringToFind) const noexcept
+{
+	if(cszStringToFind == nullptr)
+	{
+		return std::nullopt;
+	}
+	
+	size_t nStringLength = GetStringLength(cszStringToFind);
+	if(nStringLength <= 1)
+	{
+		return std::nullopt;
+	}
+	nStringLength -= 1; // Excluding terminator
+
+	size_t nIndex = 0; // Index for iterating through this.size()
+	size_t nCurrentStringIndex = 0; // Index for iterating through cszStringToFind
+	bool bIsLookingFor = false;
+
+	for(; nIndex < m_nSize; nIndex++)
+	{
+		if(m_szData[nIndex] != cszStringToFind[nCurrentStringIndex])
+		{
+			nCurrentStringIndex = 0;
+			bIsLookingFor = false;
```

Comment:
I think you can avoid this var usage in the loop
Just assign true value to this var before break in the condition

Reply:
Moved to break block